server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala

```diff
@@ -57,6 +59,36 @@ object MailboxSetRequest {
 }
 
 case class RemoveEmailsOnDestroy(value: Boolean) extends AnyVal
+
+object MailboxCreationRequest {
+  private val serverSetProperty = Set("id", "sortOrder", "role", "totalEmails", "totalThreads", "unreadEmails", "unreadThreads", "myRights")
+  private val assignableProperties = Set("name", "parentId", "isSubscribed", "rights")
+  private val knownProperties = assignableProperties ++ serverSetProperty
+
+  def validateProperties(jsObject: JsObject): Either[MailboxCreationParseException, JsObject] = {
+    val specifiedServerSetProperties = jsObject.keys.intersect(serverSetProperty)
+    val unknwonProperties = jsObject.keys.diff(knownProperties)
+
+    if (unknwonProperties.nonEmpty) {
+      return Left(MailboxCreationParseException(MailboxSetError.invalidArgument(
```

Comment:
please don't use return in scala code
you can pattern match on the pair (unknownProperties, specifiedServerSetProperties)